manage-data/data-store/data-streams/failure-store-recipes.md

```diff
@@ -0,0 +1,1154 @@
+          "data": {
+            "counter": 42
+          }
+        },
+        "executed_pipelines": [ <3>
+          "my-datastream-remediation-pipeline",
+          "my-datastream-default-pipeline"
+        ]
+      }
+    }
+  ]
+}
+```
+1. The index name has been updated.
+2. The source is as expected after the default pipeline has run.
+3. Ensure that both the new remediation pipeline and the original default pipeline have successfully run.
+
+::::
+
+::::{step} Reindex the failure documents
+Combine the remediation pipeline with the failure store query together in a [reindex operation](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-reindex) to replay the failures.
+
+```console
+POST _reindex
+{
+  "source": {
+    "index": "my-datastream-ingest-example::failures", <1>
+    "query": {
+      "bool": { <2>
+        "must": [
+          {
+            "exists": {
+              "field": "error.pipeline"
+            }
+          },
+          {
+            "match": {
+              "document.index": "my-datastream-ingest-example"
+            }
+          },
+          {
+            "match": {
+              "error.type": "illegal_argument_exception"
+            }
+          },
+          {
+            "range": {
+              "@timestamp": {
+                "gt": "2025-05-01T00:00:00Z",
+                "lte": "2025-05-17T00:00:00Z"
+              }
+            }
+          }
+        ]
+      }
+    }
+  },
+  "dest": {
+    "index": "my-datastream-ingest-example", <3>
+    "op_type": "create",
+    "pipeline": "my-datastream-remediation-pipeline" <4>
+  }
+}
+```
+1. Read from the failure store.
+2. Only reindex failure documents that match the ones we are replaying.
+3. Set the destination to the data stream the failures originally were sent to.
+4. Replace the pipeline with the remediation pipeline.
+
+```console-result
+{
+  "took": 469,
+  "timed_out": false,
+  "total": 1,
+  "updated": 0,
+  "created": 1, <1>
+  "deleted": 0,
+  "batches": 1,
+  "version_conflicts": 0,
+  "noops": 0,
+  "retries": {
+    "bulk": 0,
+    "search": 0
+  },
+  "throttled_millis": 0,
+  "requests_per_second": -1,
+  "throttled_until_millis": 0,
+  "failures": []
+}
+```
+1. The failures have been remediated.
+
+:::{tip}
+Since the failure store is enabled on this data stream, it would be wise to check it for any further failures from the reindexing process. Failures that happen at this point in the process may end up as nested failures in the failure store. Remediating nested failures can quickly become a hassle as the original document gets nested multiple levels deep in the failure document. For this reason, it is suggested to remediate data during a quiet period when no other failures are likely to arise. Furthermore, rolling over the failure store before executing the remediation would allow easier discarding of any new nested failures and only operate on the original failure documents.
```

Comment:
I think it would also be good to mention that reindexing the failed documents into the data stream does not remove the failures, so that needs to be done after the reindexing.